internal/receiver/discoveryreceiver/correlation.go

```diff
@@ -94,61 +87,43 @@ func newCorrelationStore(logger *zap.Logger, ttl time.Duration) correlationStore
 	}
 }
 
-// UpdateEndpoint will update all existing correlation timestamps and states by endpoint.ID, or
-// creates a new no-type ~singleton w/ the initial correlation info for later use in correlation creation.
-func (s *store) UpdateEndpoint(endpoint observer.Endpoint, state endpointState, observerID component.ID) {
-	defer s.endpointLocks.Lock(endpoint.ID)()
```

Comment:
Fix the map-locking mechanism along the way. This line has to be broken down to be effective. Otherwise the whole `s.endpointLocks.Lock(endpoint.ID)()` line is deffered